src/XIVLauncher.Common.Unix/Compatibility/DxvkSettings.cs

```diff
@@ -0,0 +1,142 @@
+#nullable enable
+using System;
+using System.IO;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+
+namespace XIVLauncher.Common.Unix.Compatibility;
+
+public class DxvkSettings
+{
+    public bool Enabled { get; }
+
+    public string DownloadURL { get; }
+
+    public string FolderName { get; }
+
+    public Dictionary<string, string> DxvkVars { get; }
+
+    public Dxvk.DxvkHudType DxvkHud { get; }
+
+    public Dxvk.DxvkVersion DxvkVersion { get; }
+
+    private const string ALLOWED_CHARS = "^[0-9a-zA-Z,=.]+$";
+
+    private const string ALLOWED_WORDS = "^(?:devinfo|fps|frametimes|submissions|drawcalls|pipelines|descriptors|memory|gpuload|version|api|cs|compiler|samplers|scale=(?:[0-9])*(?:.(?:[0-9])+)?)$";
+
+    public DxvkSettings(Dxvk.DxvkHudType hud, DirectoryInfo corePath, Dxvk.DxvkVersion version, bool enabled = true, string? dxvkHudCustom = null, FileInfo? mangoHudConfig = null, bool async = true,
+                        int maxFrameRate = 0)
+    {
+        Enabled = enabled;
+        DxvkHud = hud;
+        var dxvkConfigPath = new DirectoryInfo(Path.Combine(corePath.FullName, "compatibilitytool", "dxvk"));
+        if (!dxvkConfigPath.Exists)
+            dxvkConfigPath.Create();
+        DxvkVars = new Dictionary<string, string>
+        {
+            { "DXVK_LOG_PATH", Path.Combine(corePath.FullName, "logs") },
+            { "DXVK_CONFIG_FILE", Path.Combine(dxvkConfigPath.FullName, "dxvk.conf") },
+            { "DXVK_FRAME_RATE", (maxFrameRate).ToString() }
+        };
+        DxvkVersion = version;
+        var release = DxvkVersion switch
+        {
+            Dxvk.DxvkVersion.v1_10_1 => "1.10.1",
+            Dxvk.DxvkVersion.v1_10_2 => "1.10.2",
+            Dxvk.DxvkVersion.v1_10_3 => "1.10.3",
+            Dxvk.DxvkVersion.v2_0 => "2.0",
+            Dxvk.DxvkVersion.v2_1 => "2.1",
+            Dxvk.DxvkVersion.v2_2 => "2.2",
+            _ => throw new ArgumentOutOfRangeException(),
+        };
+        if (new[] {"1.10.1", "1.10.2", "1.10.3", "2.0"}.Contains(release))
+        {
+            DownloadURL = $"https://github.com/Sporif/dxvk-async/releases/download/{release}/dxvk-async-{release}.tar.gz";
+            FolderName = $"dxvk-async-{release}";
+            DxvkVars.Add("DXVK_ASYNC", async ? "1" : "0");
+        }
+        else
+        {
+            DownloadURL = $"https://github.com/doitsujin/dxvk/releases/download/v{release}/dxvk-{release}.tar.gz";
+            FolderName = $"dxvk-{release}";
+        }
+
+        DirectoryInfo dxvkCachePath = new DirectoryInfo(Path.Combine(dxvkConfigPath.FullName, "cache"));
```

Comment:
Very minor nitpick, this is a bit inconsistent with the use of var everywhere else

Reply:
Changed.